tests/identity.rs

```diff
@@ -267,3 +267,40 @@ fn refresh() {
 
     assert_eventually!(refreshed.load(Ordering::SeqCst) == true);
 }
+
+#[test]
+fn force_identity_header() {
+    let _ = env_logger_init();
+    let proxy_id = "foo.ns1.serviceaccount.identity.linkerd.cluster.local";
+    let app_id = "bar.ns1.serviceaccount.identity.linkerd.cluster.local";
+
+    let proxy_identity = identity::Identity::new("foo-ns1", proxy_id.to_string());
+    let app_identity = identity::Identity::new("bar-ns1", app_id.to_string());
+
+    let srv = server::http2_tls(app_identity.server_config)
+        .route("/", "hello")
+        .run();
+
+    let proxy = proxy::new()
+        .outbound(srv)
+        .identity(proxy_identity.service().run())
+        .run_with_test_env(proxy_identity.env);
+
+    // Make an http client with no TLS config
+    let client = client::http2(proxy.outbound, app_id);
+
+    // Send a request with a `l5d-force-id` header value of `app_id` so that
```

Comment:
I think we should also test the failure scenarios: 
* When the l5d-force-id header is present and the server's TLS id doesn't match it, the connection should fail
* When the l5d-force-id header is present and the server does not have a TLS id, the connection should fail.